LabExT/Measurements/InsertionLossSweep.py

```diff
@@ -181,7 +207,63 @@ def algorithm(self, device, data, instruments, parameters):
         self.instr_laser.close()
         self.instr_pm.close()
 
+        # apply reference to data
+        if parameters['file path to reference meas.'].value.strip():
+            self.apply_reference_to_data(data=data)
+        else:
+            data['values']['referenced transmission [dB]'] = []
+
+        # if user wants to save disk space by discarding raw data, do it now
+        if parameters['discard raw transmission data'].value:
+            data['values']['transmission [dBm]'] = []
+
         # sanity check if data contains all necessary keys
         self._check_data(data)
 
         return data
+
+    def check_if_reference_valid(self, parameters):
+
+        # load reference file
+        ref_fp = parameters['file path to reference meas.'].value.strip()
+        try:
+            with open(ref_fp, 'r') as fp:
+                ref_raw_data = json.load(fp=fp)
+        except FileNotFoundError:
+            raise FileNotFoundError(f'Reference file "{ref_fp:s}" not found.')
+        except json.decoder.JSONDecodeError:
+            raise RuntimeError(f'Reference file "{ref_fp:s}" could not be JSON decoded.')
+
+        # check reference parameters
+        if ref_raw_data['measurement name'] != self.name:
```

Comment:
Do you think this is strictly necessary? What about only checking that the parameters are present in the measurement settings (like you did below) and also checking for the correct keys in the `data['values']` dictionary? Then we could still apply this in measurements that are based on InsertionLossSweep but with additional features (like applying or optimizing for voltage before performing the sweep)

Reply:
Thank you for this valuable suggestion. Removed the check in [f2d54f0](https://github.com/LabExT/LabExT/pull/168/commits/f2d54f081abd63d36820fc78c988dd58af716940).